base_layer/core/src/blocks/block_header.rs

```diff
@@ -216,21 +217,23 @@ impl BlockHeader {
     /// Provides a hash of the header, used for the merge mining.
     /// This differs from the normal hash by not hashing the nonce and kernel pow.
     pub fn merged_mining_hash(&self) -> Vec<u8> {
-        ConsensusHasher::default()
-                .chain(&self.version)
-                .chain(&self.height)
-                .chain(&self.prev_hash)
-                .chain(&self.timestamp)
-                .chain(&self.input_mr)
-                // TODO: Cleanup if/when we migrate to fixed 32-byte array type for hashes
-                .chain(&copy_into_fixed_array_lossy::<_, 32>(&self.output_mr))
-                .chain(&self.output_mmr_size)
-                .chain(&copy_into_fixed_array_lossy::<_, 32>(&self.witness_mr))
-                .chain(&copy_into_fixed_array_lossy::<_, 32>(&self.kernel_mr))
-                .chain(&self.kernel_mmr_size)
-                .chain(&self.total_kernel_offset)
-                .chain(&self.total_script_offset)
-                .finalize().to_vec()
+        DomainSeparatedConsensusHasher::<BlocksHashDomain>::new("block_header")
+            .chain(&self.version)
+            .chain(&self.height)
+            .chain(&self.prev_hash)
+            .chain(&self.timestamp)
+            .chain(&copy_into_fixed_array::<_, 32>(self.input_mr.as_slice()).expect("Fixed array size 32 cannot fail"))
```

Comment:
Can't we just use the slice directly?
```suggestion
            .chain(self.input_mr.as_slice())
```